spec/models/user_spec.rb

```diff
@@ -1,64 +1,59 @@
 require "rails_helper"
 
 RSpec.describe User do
-  let(:valid_user_params) { attributes_for(:user) }
-  let(:valid_user) { create(:user) }
-  let(:valid_wishlist_params) { attributes_for(:wishlist) }
-
   it { is_expected.to have_many :wishlists }
 
   it "is valid with valid params" do
-    expect(User.new(valid_user_params)).to be_valid
+    expect(build(:user)).to be_valid
   end
 
   it "has associated wishlists that should be destroyed" do
-    valid_user.wishlists.create(valid_wishlist_params)
-    expect { valid_user.destroy }.to change { Wishlist.count }.by(-1)
+    wishlist = create(:wishlist)
+    expect { wishlist.user.destroy }.to change { Wishlist.count }.by(-1)
   end
 
   context "is invalid" do
     context "with an email that's" do
       it "nil" do
-        expect(User.new(valid_user_params.merge!(email: nil))).not_to be_valid
+        expect(build(:user, email: nil)).not_to be_valid
       end
 
       it "empty" do
-        expect(User.new(valid_user_params.merge!(email: ""))).not_to be_valid
+        expect(build(:user, email: "")).not_to be_valid
       end
 
       it "blank" do
-        expect(User.new(valid_user_params.merge!(email: "  "))).not_to be_valid
+        expect(build(:user, email: "  ")).not_to be_valid
       end
 
       it "mangled" do
-        expect(User.new(valid_user_params.merge!(email: "bob-example@"))).not_to be_valid
+        expect(build(:user, email: "bob-example@")).not_to be_valid
       end
 
       it "non-unique" do
-        User.create(valid_user_params)
-        expect(User.new(valid_user_params.merge!(email: valid_user_params[:email]))).not_to be_valid
+        expect(build(:user, email: create(:user).email)).not_to be_valid
```

Comment:
It does, but there's a user when you want to associate email
`email: create(:user).email`
 it creates new user but it can't associate it because there's already user with this email. I don't know if my explanation is understandable.